src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -592,6 +613,11 @@ def analyze(
             str(expr)
         )  # pragma: no cover
 
+    def internal_alias_extractor(self, expr: Expression) -> Expression:
```

Comment:
one last comment: can you add comment on when this function needs to called when analyzing expression to give some context to future us?